### ISSUE TITLE
Win10 WSL Ubuntu 20.04 Support

### ISSUE THREAD

**Issue by esmasth** [sysrepo/sysrepo#2155]
Hello, I have been trying to setup a working environment based on sysrepo-1.4.70 with netopeer2-1.1.39, libnetconf2-1.1.26, libyang-1.0.184 and libssh-0.9.2. I have hit the following snag in the process, and hopefully it is something amiss in the steps else it would be good if the support for the Win10 WSL environment can be fixed. Maybe it is an issue which is uncovered by WSL environment.

Steps (with installation via sudo) leading to netopeer2-server crash in sysrepo code

1. build and install libyang
2. build and install libnetconf2
3. build and install sysrepo
4. build and install netopeer2
5. install the attached ietf-netconf-acm.xml
6. start the netopeer2-server and connect to localhost via netopeer2-cli and run get once (or get-config twice).

netopeer2-cli runs get implicitly so I guess the issue is when we have a get or get-config twice from the perspective of the server. Note that a custom SYSREPO_SHM_PREFIX was not provided, and starting sysrepo-plugind or not between step 5-6 did not seem to make a difference.

[ietf-netconf-acm.xml.log](https://github.com/sysrepo/sysrepo/files/5254698/ietf-netconf-acm.xml.log)

[wsl.log](https://github.com/sysrepo/sysrepo/files/5254657/wsl.log)

**Comment by michalvasko** [comment]
Hi,
if you try the latest *sysrepo* you would not even manage to compile it on WSL because I got tired of these weird errors that are caused by missing function implementations. So, WSL is simply not supported. Additionally, Ubuntu 20.04 has some strange [permission problem](https://github.com/sysrepo/sysrepo/issues/2080) that we have not managed to fix/learn any details about. Meaning you should avoid using it as well. For example, you can install Ubuntu 18.04 VM on your Windows.

**Comment by esmasth** [comment]
Hi Michal, thanks for getting back on the support status. Can you you please point to the additional changes in latest after sysrepo 1.4.70 baseline which will make it impossible to compile on WSL just for my curiousity? I couldn't find it easily.

**Comment by michalvasko** [comment]
Hi,
just a simple [CMake check](https://github.com/sysrepo/sysrepo/blob/devel/CMakeLists.txt#L34) that can even be bypassed but it should make it obvious you are doing something unsupported so you will not receive any support from us.

**Comment by esmasth** [comment]
Thanks for the information and quick responses, Michal. I will close this now.

I might dig further myself for personal convenience later but anyone in the user community please drop a hint if anything as it helps for quick checks.

**Comment by nullromo** [comment]
For what it's worth, I have been compiling and running sysrepo on WSL for a few years now and it's never led to any issues. I just found out WSL is "not supported" today after I saw [this commit](https://github.com/sysrepo/sysrepo/commit/9c985454552d18678bfc2b2441edb2731c8f5f25) which "fixed" the WSL check.

Not sure what about sysrepo isn't compatible with WSL, but just thought I would mention everything seems to be OK with it for me. But I will now just have to add `-DFORCE_WSL=1` when compiling.
